src/FsAutoComplete/LspServers/AdaptiveFSharpLspServer.fs

```diff
@@ -1637,7 +1626,16 @@ type AdaptiveFSharpLspServer(workspaceLoader: IWorkspaceLoader, lspClient: FShar
 
           let doc = p.TextDocument
           let filePath = doc.GetFilePath() |> Utils.normalizePath
-          let file = volaTileFile filePath p.Text.Value None DateTime.UtcNow
+
+          let file =
+            option {
+              let! oldFile = forceFindOpenFile filePath
+              let oldFile = p.Text |> Option.map (oldFile.SetText) |> Option.defaultValue oldFile
+              return oldFile.UpdateTouched()
+            }
+            |> Option.defaultWith (fun () ->
+              // Very unlikely to get here
+              VolatileFile.Create(filePath, p.Text.Value, None, DateTime.UtcNow))
```

Comment:
Lots of fumbling to try to be safe around DidSave and updating a VolatileFile correctly